Homework/reactPro/ngo/src/components/MainContainer.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import Header from './Header';
+import Navigation from './Navigation';
+import SearchOption from './SearchOption';
+import Legend from './Legend';
+import UserSearch from './UserSearch';
+import Organisation from './Organisation';
+
+import '../App.css';
+
+
+class MainContainer extends React.Component {
+  
+  constructor(props){
+    super(props);
+    this.state={
+      pageHeader: 'London NGO Trucking System',
+      searchTypeResult: [],
+      organizationList: [],
+      searchType: []
+    };
+  }
+
+  render(){
+    return (
+      <div className="App">
+        <Navigation /> 
+        <form name="mainForm" className="group" action="#">
+        <ol>  
+          <Legend title="Advanced Search"/>
+          <SearchOption getService={this.getServiceType} title="Search option" placeHolder="Select..." 
+             type="By service type" area="By service area" clients="By service clients" />
+          <UserSearch searchTypeList={ this.state.searchTypeResult } getUserInput={this.getOrganization}/>
+        </ol>
+        <div id="searchResult" >
+          <Organisation organisations={this.state.organizationList} />
+        </div>
+      </form>   
+      </div>
+    );
+  }
+
+  componentDidMount=()=>{
+     
+  };
+
+  getServiceType=(Type)=>{
+    const url='https://code-your-future.github.io/api-demo/' + Type + '/index.json';
+    this.setState({ searchType: Type });
+    this.getServiceTypeDetailAPI(url);
+        
+  };
+  
+  getOrganization=(userInput)=>{
+    const url='https://code-your-future.github.io/api-demo/'+ this.state.searchType +'/' + userInput +'/index.json';
+    this.getOrganizationDetailAPI(url);   
+    //console.log(this.state.organizationList);  
+  };
+
+  getOrganizationDetailAPI = (url) => {
+    const APIAddress = url;
+    fetch(APIAddress)
+    .then(response => {
+      return response.json();
+    })
```

Comment:
Consider what would happen here if the fetch wasn't successful or returned something other than a 200 response. This would cause this to pass through some non-existant data through to the next `.then()` block.

You might want to consider having a check to ensure the response was good. 

Something like 
```js
fetch(url)
  .then(response => {
    if (response.ok) {
      return response.json();
    } else {
      console.error(`API fetch failed with ${response.status}`);
    }
  })
  .then(jsonData => {
    this.setState( { organizationList: jsonData.data } );
  })
```

You could also throw a `new Error` and add a `.catch` block to deal with that. You should do this anyway as potentially your code will error on Line 68 if it tried to read `data` of an `undefined` value.